### PR TITLE
chore(react-chart): add zoom-and-pan demo

### PR DESCRIPTION
**Pull request by DmitryBogomolov**

### CODE REVIEW

File: site/_data/demos/navigation.yml [hunk 1]
```diff
@@ -74,6 +74,10 @@
     items:
       - title: Multiple Axes
         path: /react/chart/demos/combination/multiple-axes
+  - title: Zoom and Pan
+    items:
+      - title: Zoom and Pan
+        path: /react/chart/demos/zoom-and-pan/scatter
```
Comment (Krijovnick) [inline site/_data/demos/navigation.yml]: The demo is not working on the site. The folder with name which contain hyphens is not parsing correctly

File: packages/dx-react-chart-demos/src/demo-sources/zoom-pan/bootstrap4/scatter.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import { dataGenerator } from '../../../demo-data/generator';
+
+const adjustDomain = ([start, end]) => [Math.floor(start), Math.ceil(end)];
+
+const formatTooltipValue = format('.2f');
+const TooltipContent = ({ targetItem, data, ...restProps }) => {
+  const item = data[targetItem.point];
+  const arg = item[targetItem.series === 'Series 1' ? 'arg1' : 'arg2'];
+  const val = item[targetItem.series === 'Series 1' ? 'val1' : 'val2'];
+  const text = `${formatTooltipValue(arg)} | ${formatTooltipValue(val)}`;
```
Comment (Krijovnick) [inline packages/dx-react-chart-demos/src/demo-sources/zoom-pan/bootstrap4/scatter.jsx]: It seems to me the text of tooltip is hard to understand. I would expect to see something like this: 
`argument: 0.77`
`value: 0.25`

File: packages/dx-react-chart-demos/src/demo-sources/zoom-pan/bootstrap4/scatter.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            <EventTracker />
+            <HoverState />
+            <Tooltip contentComponent={this.TooltipContent} />
+
+            <ZoomAndPan
+              viewport={viewport}
+              onViewportChange={this.changeViewport}
+              interactionWithValues="both"
+            />
+          </Chart>
```
Comment (Krijovnick) [inline packages/dx-react-chart-demos/src/demo-sources/zoom-pan/bootstrap4/scatter.jsx]: Why there is no animation?

File: site/_data/demos/navigation.yml [hunk 4]
```diff
@@ -74,6 +74,10 @@
     items:
       - title: Multiple Axes
         path: /react/chart/demos/combination/multiple-axes
+  - title: Zoom and Pan
```
[the remark this answers is not in the document]
Comment (DmitryBogomolov) [reply site/_data/demos/navigation.yml]: Updated.

File: site/_data/demos/navigation.yml (hunk 1 above)
Comment (DmitryBogomolov) [reply site/_data/demos/navigation.yml]: Updated.
(though locally site was successfully build)

File: packages/dx-react-chart-demos/src/demo-sources/zoom-pan/bootstrap4/scatter.jsx (hunk 2 above)
Comment (DmitryBogomolov) [reply packages/dx-react-chart-demos/src/demo-sources/zoom-pan/bootstrap4/scatter.jsx]: Update. Though I used *X* and *Y* for better alignment.

File: packages/dx-react-chart-demos/src/demo-sources/zoom-pan/bootstrap4/scatter.jsx (hunk 3 above)
Comment (DmitryBogomolov) [reply packages/dx-react-chart-demos/src/demo-sources/zoom-pan/bootstrap4/scatter.jsx]: Added.

**Pull request merged**